app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -33,19 +39,14 @@ class SearchRepositoriesViewModel(private val repository: GithubRepository) : Vi
         private const val VISIBLE_THRESHOLD = 5
     }
 
-    private val queryLiveData = MutableLiveData<String>()
-    val repoResult: LiveData<RepoSearchResult> =  queryLiveData.switchMap {
-        liveData {
-            val repos = repository.getSearchResultStream(it).asLiveData(Dispatchers.Main)
-            emitSource(repos)
-        }
-    }
-
+    private var lastQueryValue: String? = null
```

Comment:
I imagine this needs to be volatile as it can access by different threads

Reply:
These look to be always accessed on the UI thread, at least as the code is now - searchRepo is only ever called in one place from within a viewModelScope, which is main thread: https://developer.android.com/reference/kotlin/androidx/lifecycle/package-summary#viewmodelscope